nexus/src/db/error.rs

```diff
@@ -80,64 +80,52 @@ pub fn diesel_pool_result_optional<T>(
     }
 }
 
-/// Converts a Diesel pool error to an external error when operating on a
-/// particular resource, identified by resource type and lookup type
-/// NOTE: If you already have an `authz::ApiResource`, you should use
-/// [`public_error_from_diesel_pool_authz()`] instead.  Eventually, the only
-/// uses of this function should be in the DataStore functions that actually
-/// look up a record for the first time.
-pub fn public_error_from_diesel_pool_lookup(
-    error: PoolError,
-    resource_type: ResourceType,
-    lookup_type: LookupType,
-) -> PublicError {
-    public_error_from_diesel_pool_helper(error, |error| {
-        public_error_from_diesel_lookup_helper(error, || {
-            PublicError::ObjectNotFound {
-                type_name: resource_type,
-                lookup_type,
-            }
-        })
-    })
-}
-
-/// Converts a Diesel pool error to an external error when operating on a
-/// particular resource, identified by ApiResource
-pub fn public_error_from_diesel_pool_authz<R>(
-    error: PoolError,
-    resource: &R,
-) -> PublicError
-where
-    R: crate::authz::ApiResource,
-{
-    public_error_from_diesel_pool_helper(error, |error| {
-        public_error_from_diesel_lookup_helper(error, || resource.not_found())
-    })
-}
-
-/// Converts a Diesel pool error to an external error,
-/// when requested as part of a creation operation
-pub fn public_error_from_diesel_pool_create(
-    error: PoolError,
-    resource_type: ResourceType,
-    object_name: &str,
-) -> PublicError {
-    public_error_from_diesel_pool_helper(error, |error| {
-        public_error_from_diesel_create(error, resource_type, object_name)
-    })
+/// Describes the operation type which triggered the error,
+/// as well as context which may be used to populate more informative errors.
+pub enum OpKind<'a> {
```

Comment:
Yeah, I like short names that are disambiguated by module - but actually, the `OpKind` in `OpContext` isn't used outside that module, and it is not visible to `datastore.rs`.

I made it non-`pub`, since it does not need to be - now there's only one `OpKind` that even could be used within the `db` module. If anything in the `db` module needed to deal with multiple `OpKind` objects, I'd definitely agree to have distinct naming conventions, but this is kinda why I like keeping things module-scoped -- names can remain short until there's a reason to make them longer.

Reply:
> Yeah, I like short names that are disambiguated by module - but actually, the OpKind in OpContext isn't used outside that module, and it is not visible to datastore.rs.

This is true, but developers still have the cognitive overhead of two completely unrelated things called `OpKind`.  (Not saying we need to change it though!)